src/cargo/sources/git/utils.rs

```diff
@@ -370,6 +370,14 @@ impl<'a> GitCheckout<'a> {
                 anyhow::format_err!("non-utf8 url for submodule {:?}?", child.path())
             })?;
 
+            // Skip the submodule if the config says not to update it.
+            if child.update_strategy() == git2::SubmoduleUpdate::None {
+                cargo_config
+                    .shell()
+                    .status("Skipping", format!("git submodule `{}`", url))?;
```

Comment:
I am wondering if this is the correct status message. I feel as though it might be confusing as to why it is not being updated. Stating why it is not being updated might add clarity to what is going on. 

I would hold off on doing what I suggested until one of the maintainers looks at this as they might not agree with what I suggested

Reply:
Happy to tweak it. Say something like
```
Skipping git submodule [..] due to its update strategy
```
I wasn't feeling very creative and copied it from `git submodule` UI.